src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientHandler.java

```diff
@@ -50,6 +63,25 @@ public static boolean isClientValid(SnowflakeStreamingIngestClient client) {
     return client != null && !client.isClosed() && client.getName() != null;
   }
 
+  /**
+   * Gets the loggable properties (see {@link
+   * StreamingClientHandler#LOGGABLE_STREAMING_CONFIG_PROPERTIES} passed into the client.
+   *
+   * @param properties The client properties
+   * @return A string with the loggable properties
+   */
+  public static String getLoggableClientProperties(Properties properties) {
+    return properties == null
+        ? ""
+        : properties.entrySet().stream()
+            .filter(
+                propKvp ->
+                    LOGGABLE_STREAMING_CONFIG_PROPERTIES.stream()
+                        .anyMatch(propKvp.getKey().toString()::equalsIgnoreCase))
+            .collect(Collectors.toList())
+            .toString();
```

Comment:
are we fetching corresponding values?

Reply:
yup, theres a test that confirms this. the filter is only looking at the key but it also returns the value in the .collect

Reply:
got it thanks!